stdlib/http/src/main/java/org/ballerinalang/net/http/WebSocketObservability.java

```diff
@@ -0,0 +1,387 @@
+import org.slf4j.LoggerFactory;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static org.ballerinalang.jvm.observability.ObservabilityConstants.SERVER_CONNECTOR_WEBSOCKET;
+
+/**
+ * Providing observability functionality to WebSockets.
+ *
+ * @since 1.1.0
+ */
+public class WebSocketObservability {
+
+    private static final Logger logger = LoggerFactory.getLogger(WebSocketObservability.class);
+
+    //Observability Constants
+    private static final String TAG_CONNECTION_ID = "connectionID";
+    private static final String TAG_KEY_RESULT = "result";
+    private static final String TAG_CLIENT_OR_SERVER = "client_or_server";
+    private static final String TAG_SERVICE = "service";
+    private static final String TAG_MESSAGE_TYPE = "type";
+    private static final String TAG_ERROR_TYPE = "error_type";
+
+    private static final String METRIC_REQUESTS = "requests";
+    private static final String METRIC_REQUESTS_DESC = "Number of WebSocket connection requests";
+
+    private static final String METRIC_CONNECTIONS = "connections";
+    private static final String METRIC_CONNECTIONS_DESC = "Number of currently active connections";
+
+    private static final String METRIC_MESSAGES_RECEIVED = "messages_received";
+    private static final String METRIC_MESSAGES_RECEIVED_DESC = "Number of messages received";
+
+    private static final String METRIC_MESSAGES_SENT = "messages_sent";
+    private static final String METRIC_MESSAGES_SENT_DESC = "Number of messages sent";
+
+    public static final String WEBSOCKET_MESSAGE_RESULT_SUCCESS = "success";
+
+    private static final String METRIC_ERRORS = "errors";
+    private static final String METRIC_ERRORS_DESC = "Number of errors";
+
+    private static final String WEBSOCKET_CLIENT_OR_SERVER_CLIENT = "client";
+    private static final String WEBSOCKET_CLIENT_OR_SERVER_SERVER = "server";
+
+    public static final String WEBSOCKET_MESSAGE_TYPE_TEXT = "text";
+    public static final String WEBSOCKET_MESSAGE_TYPE_BINARY = "binary";
+    public static final String WEBSOCKET_MESSAGE_TYPE_CONTROL = "control";
+    public static final String WEBSOCKET_MESSAGE_TYPE_CLOSE = "close";
+
+    static final String WEBSOCKET_ERROR_TYPE_CONNECTION = "connection";
+    public static final String WEBSOCKET_ERROR_TYPE_CLOSE = "close";
+    public static final String WEBSOCKET_ERROR_TYPE_MESSAGE_SENT = "message_sent";
+    static final String WEBSOCKET_ERROR_TYPE_MESSAGE_RECEIVED = "message_received";
+    static final String WEBSOCKET_ERROR_TYPE_UNEXPECTED = "unexpected";
+
+    private static final String WEBSOCKET_UNKNOWN = "unknown";
+
+    /**
+     * This method observes all incoming WebSocket connection requests.
+     *
+     * @param connectionInfo information regarding connection.
+     * @param result whether the connection was successful or not.
+     */
+    static void observeRequest(WebSocketOpenConnectionInfo connectionInfo, String result) {
```

Comment:
I don't see you using this function to observe unsuccessful requests? If you are not using it do we need the result parameter?